src/middlewared/middlewared/plugins/api_key.py

```diff
@@ -1,17 +1,22 @@
-import random
-import string
-from typing import Literal, TYPE_CHECKING
+import pam
+import errno
 
-from passlib.hash import pbkdf2_sha256
+from typing import Literal, TYPE_CHECKING
 
+from datetime import datetime, UTC
 from middlewared.api import api_method
 from middlewared.api.current import (
     ApiKeyEntry, ApiKeyCreateArgs, ApiKeyCreateResult, ApiKeyUpdateArgs, ApiKeyUpdateResult,
-    ApiKeyDeleteArgs, ApiKeyDeleteResult, HttpVerb,
+    ApiKeyDeleteArgs, ApiKeyDeleteResult, ApiKeyMyKeysArgs, ApiKeyMyKeysResult,
 )
-from middlewared.service import CRUDService, private, ValidationErrors
+from middlewared.service import CRUDService, pass_app, private, ValidationError, ValidationErrors
+from middlewared.service_exception import CallError
 import middlewared.sqlalchemy as sa
-from middlewared.utils.allowlist import Allowlist
+from middlewared.utils import filter_list
+from middlewared.utils.auth import LEGACY_API_KEY_USERNAME
+from middlewared.utils.crypto import generate_pbkdf2_512, generate_string
+from middlewared.utils.privilege import credential_has_full_admin
+from middlewared.utils.sid import sid_is_valid
 from middlewared.utils.time_utils import utc_now
 if TYPE_CHECKING:
     from middlewared.main import Middleware
```

Comment:
Middleware is unused, but this might be unavoidable.  The flake8 error is `F401`. 
Maybe add `# noqa: F401`